source/tomopy/misc/corr.py

```diff
@@ -267,6 +268,137 @@ def median_filter_cuda(arr, size=3, axis=0):
+
+    y_idx :  int
+        The y index for a given nonfinite value.
+
+    shape_x : int
+        The max shape of the 2D projection in the x dimension.
+
+    shape_y : int
+        The max shape of the 2D projection in the y dimension.
+
+    Returns
+    -------
+    The integer kernel bounds to surround the nonfinite value with.
+    """
+
+    # Determining x kernel
+    x_idx_lower = x_idx - kernel
+    x_idx_higher = x_idx + kernel + 1
+
+    if x_idx_lower < 0:
+        x_idx_lower = 0
+    if x_idx_higher > shape_x-1:
+        x_idx_higher = shape_x-1
+
+    # Determining y kernel
+    y_idx_lower = y_idx - kernel
+    y_idx_higher = y_idx + kernel + 1
+
+    if y_idx_lower < 0:
+        y_idx_lower = 0
+    if y_idx_higher > shape_y-1:
+        y_idx_higher = shape_y-1
+
+    # Forcing values to be integers
+    integer_values_pre = [x_idx_lower, x_idx_higher, y_idx_lower, y_idx_higher]
+    integer_values = [int(i) for i in integer_values_pre]
+
+    return integer_values
+
+
+def median_filter_nonfinite(data, kernel=1):
+    """Apply a selective median filter with size kernel to all nonfinite
+       values in the 3D data array.
+
+    Parameters
+    ----------
+    data : 3D nd.array
+        The 3D array of data with nonfinite values in it.
+
+    kernel : int
+        The size of the kernel to be used for a local median filter. 
+
+    Returns
+    -------
+    The corrected 3D array with all nonfinite values removed based upon the local
+    median value defined by the kernel size.
+    """
+
+    # Creating store arrays for the prj, x, y indicies for bad values
+    nonfinite_idx_store = [[], [], []]
+
+    # Iterating throug each projection to save on RAM
+    for idx, projection in tqdm(enumerate(data), desc='Finding Bad Values', total=len(data)):
+        nonfinite_idx_store = np.concatenate((nonfinite_idx_store,
+                                              find_nonfinite_values(projection, int(idx))), axis=1)
+
+    shape = np.shape(data)
+
+    # Iterating through each bad value and replace it with finite median
+    for indices in tqdm(zip(nonfinite_idx_store.transpose()), desc='Removing Bad Values'):
+
+        # Turning index values to integers
+        prj_idx, x_idx, y_idx = [int(ii) for ii in indices[0]]
+
+        # Determining the lower and upper bounds for kernel
+        x_lower, x_higher, y_lower, y_higher = determine_nonfinite_kernel_idxs(x_idx,
+                                                                               y_idx,
+                                                                               kernel,
+                                                                               shape[1],
+                                                                               shape[2])
+
+        # Extracting kernel data and fining finite median
+        kernel_cropped_data = data[prj_idx, x_lower:x_higher, y_lower:y_higher]
+        kernel_cropped_data = np.asarray(kernel_cropped_data)
```

Comment:
A slice of data is already a numpy array? Why do we need to ensure that it is an array?
```suggestion
```

Reply:
This has been removed in the latest commit (040be55 - Updating median_filter_nonfinite to not require tqdm)